src/dataset-builder/CohortEditor.ts

```diff
@@ -478,18 +500,18 @@ const CohortEditorContents: React.FC<CohortEditorContentsProps> = (props) => {
 
 interface CohortEditorProps {
   readonly onStateChange: OnStateChangeHandler;
-  readonly snapshotBuilderSettings: SnapshotBuilderSettings;
+  readonly dataset: DatasetModel;
```

Comment:
I like the idea of using `DatasetModel` instead of settings here, but I don't like that using it means that we have to check when using `snapshotBuilderSettings`; this code should only run if `snapshotBuilderSettings` is defined. Would it be possible to create a new `DatasetModel` type where `snapshotBuilderSettings` is always defined? Maybe using a subtype; not sure what the best TS way to do that would be.

Reply:
Hmmm... I think I have an approach but I don't know if I like it as much. I'll push it and see what you think.

Reply:
Why are we changing to DatasetModel here? Is there more information we need from the dataset that isn't included in the  snapshotBuilderSettings? Also will snapshotBuilderSettings ever not exist on a dataset/what happens then?

Reply:
Discussed offline! Has to do with availability of the id.